src/iuse.cpp

```diff
@@ -4332,15 +4278,25 @@ int iuse::circsaw_on(player *p, item *it, bool t, const tripoint& )
 
 int iuse::jackhammer(player *p, item *it, bool, const tripoint &pos )
 {
+    bool normal_language = it->type->id != "jacqueshammer";
+      // Jacqueshammers function the same as ordinary
+      // jackhammers, except they print messages in French for
+      // comic effect.
     if (it->charges < it->type->charges_to_use()) {
         return 0;
     }
     if (p->is_underwater()) {
-        p->add_msg_if_player(m_info, _("You can't do that while underwater."));
+        p->add_msg_if_player(m_info, normal_language
+          ? _("You can't do that while underwater.")
+          : _("Vous ne pouvez pas faire que sous l'eau."));
```

Comment:
I'd just like to chip in something that I've been thinking about
regarding iUse logic. We've been wanting to get more use out of the
LUA implementation, why not start migrating all the iUse stuff to lua,
filling in the API as needed to flesh it out more?

On 6/13/15, Kevin Granade <notifications@github.com> wrote:
>>      if (it->charges < it->type->charges_to_use()) {
>>          return 0;
>>      }
>>      if (p->is_underwater()) {
>> -        p->add_msg_if_player(m_info, _("You can't do that while
>> underwater."));
>> +        p->add_msg_if_player(m_info, normal_language
>> +          ? _("You can't do that while underwater.")
>> +          : _("Vous ne pouvez pas faire que sous l'eau."));
>
> Yea just like that.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/CleverRaven/Cataclysm-DDA/pull/12631/files#r32372450